### PR TITLE
add support for german banks, for example Sparkasse

### PR DESCRIPTION
**Pull request by tpokorra**

### CODE REVIEW

File: src/Kingsquare/parser/banking/mt940/engine/spk.php [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+	/**
+	 * Overloaded: Sparkasse does not have a header line
+	 * @inheritdoc
+	 */
+	protected function parseStatementData() {
+		$results = preg_split('/(^:20:|^-X{,3}$|\Z)/sm',
+				$this->getRawData(),
+				-1,
+				PREG_SPLIT_NO_EMPTY);
+		//array_shift($results); // remove the header
```
Comment (fruitl00p) [inline src/Kingsquare/parser/banking/mt940/engine/spk.php]: Is this intentional? Ifso, could you remove this line as it might lead to confusion later ;)

**Comment by fruitl00p** [comment]
Cool! Thanks for the addition! (and providing a simple example unit-test) I have zero experience with german banks / german mt940 but is this really Sparkasse only?

**Comment by tpokorra** [comment]
I have removed the commented line. 

This also works for other german banks. The Evangelische Bank uses the same format (https://www.eb.de/). I only have to parse files from Sparkasse and Evangelische Bank, but I think others might be similar.

One extension would be the support of the umlaut characters, which is quite a mess in the german bank statement files. Not sure when I will find the time.

I was glad I found your project. I parse the MT940 files in C#, but had a bug this morning, and it was good to be able to compare the results of your parser with mine :)
I had a problem with a storno transaction: https://github.com/SolidCharity/OpenPetraPlugin_BankimportMT940/commit/c80b6399d35d9aa2212ccbce8daf9b96dde8195f

**Comment by fruitl00p** [comment]
Cool!

Glad you found it useful and found the time for this pull! :+1:

**Pull request merged**